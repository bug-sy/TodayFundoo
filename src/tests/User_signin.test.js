// import React from 'react';
// import { render } from '@testing-library/react';
// import UserSignin from '../components/User_signin';
// import { shallow } from 'enzyme';

// test('User Signin Form', () => {
//     const component = shallow(<UserSignin/>);
//     const form = component.find('text');
//     // when
//     form.props().onChange({target: {
//        name: 'email',
//        value: 'rakesh.sahu994@gmail.com'
//     }});
//     // then
//     expect(component.state('text')).toEqual('rakesh.sahu994@gmail.com');
// });


import React from 'react';
import { shallow } from 'enzyme';
import UserSignin from '../components/User_signin';
import TextField from "@material-ui/core/TextField";


describe("MyComponent", () => {
    it("should create an entry in component state", () => {
        // given
        const component = shallow(<UserSignin />);
        const form = component.find('input');
        // when
        form.props().onChange({target: {
           name: 'myName',
           value: 'myValue'
        }});
        // then
        expect(component.state('input')).toEqual('myValue');
    });

    it("should check for onChange method (2)", () => {
        // const wrapper = shallow(<Comment onChange={}/>)
      
    
        const component = shallow(
          <UserSignin  commentBody={"test"} />
        );
        const form = component
          .find(TextField)
          .at(0)
          .props()
          .onChange({target: {
            name: 'email',
            value: 'rakesh.sahu694@gmail.com'
          }});
          
          // th
          expect(component.state('email')).toEqual('rakesh.sahu694@gmail.com');
      });

      it("should check for onChange method (3)", () => {
        const component = shallow(
          <UserSignin  commentBody={"test"} />
        );
        const form = component
          .find(TextField)
          .at(1)
          .props()
          .onChange({target: {
            name: 'password',
            value: 'abcdefg'
          }});
          
          // th
          expect(component.state('passwordTest')).toEqual('abcdefg');
       });
})