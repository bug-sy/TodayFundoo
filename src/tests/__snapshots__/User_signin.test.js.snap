// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Signin Form 1`] = `
<div
  class="main_page"
>
  <div
    class="icon"
  >
    <form
      class=""
    >
      <div>
        <div
          class="sign_in_button"
        >
          <h1>
             Sign In 
          </h1>
        </div>
        <div
          class="sign_up_button"
        >
          <h4>
             Use your google account 
          </h4>
        </div>
        <div>
          <div
            class="MuiFormControl-root MuiTextField-root text MuiFormControl-marginNormal"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
              data-shrink="false"
            >
              Email
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                name="email"
                type="Email"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-3"
                >
                  <span>
                    Email
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <br />
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              variant="contained"
            >
              Email can contain period.
            </div>
          </div>
        </div>
        <div>
          <div
            class="MuiFormControl-root MuiTextField-root text MuiFormControl-marginNormal"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
              data-shrink="false"
              for="standard-password-input"
              id="standard-password-input-label"
            >
              Password
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            >
              <input
                aria-invalid="false"
                autocomplete="current-password"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                id="standard-password-input"
                name="password"
                type="password"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-3"
                >
                  <span>
                    Password
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="sign_in_button"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Sign In
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="sign_in_button"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            href="/Signup"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              Sign Up
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
        <div
          class="sign_up_button"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            href="/sign_up"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
                variant="contained"
              >
                Reset Password ?
              </div>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
    </form>
  </div>
</div>
`;
